src/rgw/rgw_asio_frontend.cc

```diff
@@ -370,6 +374,24 @@ static int drop_privileges(CephContext *ctx)
   return 0;
 }
 
+template <typename Iter>
+int AsioFrontend::process_ports(const Iter& begin,const Iter& end,
```

Comment:
can pass iterators by value. missing space after ,

Reply:
why -should- we prefer to pass iterators by value, in general?

Reply:
in general, because that's how the standard library containers and algorithms do it. i set up an example using multimap iterators at https://www.godbolt.org/z/M_ZAfS, and with -O1 or higher there was no difference. at -O0, the `const Iter&` version generated 3 extra intructions before the call:
```asm
        mov     QWORD PTR [rbp-40], rax
        lea     rdx, [rbp-48]
        lea     rax, [rbp-40]
        mov     rsi, rdx
        mov     rdi, rax
```
vs.
```asm
        mov     rsi, rbx
        mov     rdi, rax
```

Reply:
done

Reply:
thanks, that's very helpful to know